pyop2/base.py

```diff
@@ -3856,24 +3856,25 @@ def _cache_key(cls, kernel, itspace, *args, **kwargs):
                     idx = (arg.idx.__class__, arg.idx.index)
                 else:
                     idx = arg.idx
-                map_arity = arg.map.arity if arg.map else None
+                map_offset = tuplify(arg.map.offset) if arg.map else None
                 if arg._is_dat_view:
                     view_idx = arg.data.index
                 else:
                     view_idx = None
-                key += (arg.data.dim, arg.data.dtype, map_arity,
+                key += (arg.data.dim, arg.data.dtype, map_offset,
```

Comment:
If the map is not extruded, and therefore `arg.map.offset` is `None` this key no longer treats maps with different arities as different.